kornia/contrib/connected_components.py

```diff
@@ -48,11 +49,22 @@ def connected_components(image: Tensor, num_iterations: int = 100) -> Tensor:
 
     # allocate the output tensors for labels
     B, _, _, _ = image_view.shape
-    out = torch.arange(B * H * W, device=image.device, dtype=image.dtype).view((-1, 1, H, W))
+    out = torch.arange(1, B * H * W + 1, device=image.device, dtype=image.dtype)
```

Comment:
actually now comparing to the original implementation, i'm realizing that this might be wrong for batched images, i believe that should be 
```python
out = torch.zeros(1, H, W, ...)
out = out.repeat(B, 1, 1, 1)
out[..., 0, 1] = 1.
```
and i still believe that this function we should revert to non-batched or handle batches via vmap

Reply:
I don't understand this change, moreover I don't understand why batches should be treated in a special way with vamp. The current code should work correctly with batched tensors.

Reply:
sorry for long response -- the main issue here is that with the current batched implementation i don't think we can assure that the algorithm can meet the convergence condition per each sample of the batch independently. After some digging not even sure that vmap solve that problem. So my suggestion would be to revert the base algorithm to single image and to keep backward compatibility for the batched version we just python for_loop. I'll think in the future how to improve this from Rust side.